### ISSUE TITLE
Dash is not supported in profile names

### ISSUE THREAD

**Issue by hzariv** [spring-projects/spring-boot#8782]
In my company we have an environment called "Pre-Production" and when I create an application properties file called: **application-Pre-Production.properties** it is not recognized and used by Spring Boot even though the active profile is set to spring.profiles.active=Pre-Production

I am using Spring Boot 1.5.1.RELEASE

**Comment by snicoll** [comment]
@hzariv that works for me so you'll have to share a sample that demonstrates the problem.

**Comment by hzariv** [comment]
@snicoll Sorry, It works with Spring Boot sample application. It must have been something we have done. Closing this issue. Thanks for looking into it.

**Comment by ajainy** [comment]
Facing same issue (using 1.5.2.RELEASE). @hzariv or anyone, any suggestions. 
I am using mac, and if i set profile having single word, it works. but having dash, it doesn;t. 

Example: export SPRING_PROFILES_ACTIVE=qa-03-06   doesn't work
but export SPRING_PROFILES_ACTIVE=staging  works.

**Comment by wilkinsona** [comment]
@ajainy Please see above. As far as we know, this works. If you believe that’s not the case, please provide a sample application that shows the problem and we’ll take a look.

**Comment by ajainy** [comment]
@wilkinsona Thanks for reply. I am still experimenting. 
I tried using double quotes like this `export SPRING_PROFILES_ACTIVE="qa-03-06"`. That seems to be working. But i need to do more testing. 

We using cloudfoundry for deployments, there it is working without quotes but on my local mac, it's not. It could be shell issue something. I debugged all the code from spring framework, and yes you are right, there is no where it is discriminating against dash.
